### ISSUE TITLE
Support Django 1.7

### ISSUE THREAD

**Issue by jorgecarleitao** [notanumber/xapian-haystack#144]
I tried running the tests in Django 1.7 and it does not work.

In principle this is only a [dependency on haystack](https://github.com/toastdriven/django-haystack/issues/1125), but in any case is desirable for this project.

**Comment by ikks** [comment]
I am running xapian-haystack with Django 1.7 in python 2.7 , the only thing we had problems with was https://github.com/toastdriven/django-haystack/issues/1097

**Comment by ikks** [comment]
Hello, examining haystack and Django I tried the master both for django-haystack and xapian-haystack and when trying to regenerate the index I got an `update() got an unexpected keyword argument 'commit'`.

The full backtrace is 
```
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/django/core/management/__init__.py", line 385, in execute_from_command_line
    utility.execute()
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/django/core/management/__init__.py", line 377, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/django/core/management/base.py", line 288, in run_from_argv
    self.execute(*args, **options.__dict__)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/django/core/management/base.py", line 338, in execute
    output = self.handle(*args, **options)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/haystack/management/commands/rebuild_index.py", line 26, in handle
    call_command('update_index', **options)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/django/core/management/__init__.py", line 115, in call_command
    return klass.execute(*args, **defaults)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/django/core/management/base.py", line 338, in execute
    output = self.handle(*args, **options)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/haystack/management/commands/update_index.py", line 195, in handle
    return super(Command, self).handle(*items, **options)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/django/core/management/base.py", line 503, in handle
    label_output = self.handle_label(label, **options)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/haystack/management/commands/update_index.py", line 200, in handle_label
    self.update_backend(label, using)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/haystack/management/commands/update_index.py", line 245, in update_backend
    do_update(backend, index, qs, start, end, total, verbosity=self.verbosity, commit=self.commit)
  File "/Users/igortamara/.virtualenvs/ge_web_app/lib/python2.7/site-packages/haystack/management/commands/update_index.py", line 91, in do_update
    backend.update(index, current_qs, commit=commit)
TypeError: update() got an unexpected keyword argument 'commit'
```
Maybe there is a change in the Haystack API that needs to be resolved in order to be synced.

**Comment by jorgecarleitao** [comment]
Fixed by 89f05790b19f58f7f4e9a2dbf168df2aebea6beb and tested in 767458509cc09e940a2a9556a002c1b72f331a33.
